modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/wstep/util/KeyStoreGenerator.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.mdm.mobileservices.windows.services.wstep.util;
+
+import org.wso2.carbon.mdm.mobileservices.windows.common.exceptions.KeyStoreGenerationException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.mdm.mobileservices.windows.common.Constants;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+
+/**
+ * Class for MDM Keystore operations.
+ */
+public class KeyStoreGenerator {
+
+	private static final Log logger = LogFactory.getLog(KeyStoreGenerator.class);
+
+	/**
+	 * This method loads the MDM keystore.
+	 *
+	 * @param keyStore   - MDM Keystore
+	 * @param keyStorePassword  - Keystore Password
+	 * @param keyStorePath - Keystore path
+	 * @throws KeyStoreGenerationException
+	 */
+	public static void loadToStore(KeyStore keyStore, char[] keyStorePassword, String keyStorePath)
+			throws KeyStoreGenerationException {
+
+		FileInputStream fileInputStream = null;
+
+		try {
+			if (keyStorePath != null) {
+				fileInputStream = new FileInputStream(keyStorePath);
+				keyStore.load(fileInputStream, keyStorePassword);
+			}
+		} catch (NoSuchAlgorithmException e) {
+			String msg = "Requested cryptographic algorithm is not available in the environment.";
```

Comment:
Log and throw is generally considered an anti-pattern. You can perhaps just throw the exception without logging in instances similar to this.